include/utility.hpp

```diff
@@ -453,21 +441,10 @@ namespace giml {
         }
 
         //Iterator operators to support range-based for loop syntax
-        T* begin() {
-            return this->pBackingArr;
-        }
-
-        const T* begin() const {
-            return this->pBackingArr;
-        }
-
-        T* end() {
-            return this->pBackingArr + this->length;
-        }
-
-        const T* end() const {
-            return this->pBackingArr + this->length;
-        }
+        T* begin() {return this->pBackingArr;}
```

Comment:
This absolutely reduces readability. Personally, I prefer an entire newline block and spacing here because there is logic happening, it's not just a simple `if (1) { return true; }`, rather, there is pointer logic and having them all clumped up together makes more prone for errors in pointer logic.

Reply:
Considering that these operators are simply copying the functionality from `std::vector` that people will already be familiar with, I feel condensing them is best. People can simply glance at this block and say "oh great, I know these"

Reply:
I see, I think the spacing helps too so I'm fine with it now